examples/example.cpp

```diff
@@ -4,43 +4,108 @@
 // Distributed under the Boost Software License, Version 1.0. (See accompanying
 // file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 //
+//[ main
+#include <boost/system/error_code.hpp>
 
-#include <nudb/nudb.hpp>
-#include <cstddef>
-#include <cstdint>
+#include <nudb/xxhasher.hpp>    // xxhasher
+#include <nudb/create.hpp>      // create
+#include <nudb/store.hpp>       // store
+#include <nudb/file.hpp>        // path_type
+
+#include <iostream>
+#include <cstdint>              // std::uint32_t, std::uint64_t
+
+// NuDB functions may return error_code defined by
+bool is_success(const boost::system::error_code & ec){
+    return
+        // boost system library
+        ec == boost::system::errc::success
+        // or nudb error header
+        || ec == nudb::error::success;
```

Comment:
Whoops, `success` should not be part of `nudb::error`. I think I did that to bump code coverage in some switch statmenet. The function `is_success` should never be needed, you must always be able to write `if(! ec)` to mean "if no error." I will open an issue to correct this.